### PR TITLE
Added Basic Nav with Basic Styling

### PR DESCRIPTION
**Pull request by shellycs50**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,34 +1,23 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
+import { BrowserRouter, Route, Routes } from 'react-router-dom'
 import './App.css'
-
+import Nav from './components/Nav/Nav'
+import Home from './components/home/home'
+import Error from './components/Error/Error'
```
Comment (Geoff10) [inline src/App.jsx:5]: Must: The casing on the filenames here is inconsistent, please could the Home import be updated to match the others?

File: src/App.jsx [hunk 2]
```diff
@@ -1,34 +1,23 @@
...
+   <Nav />
+   <Routes>
+    <Route path='/' element={<Home />} />
+    <Route path='*' element={<Error />} />
+   </Routes>
+   
+   
+   
+   
+   </BrowserRouter>
```
Comment (Geoff10) [inline src/App.jsx:20]: Should: The indentation needs to be fixed here to improve the code readability.
Also worth considering removing some of the extr new lines for the same reason.

File: src/components/Booklisting/Booklisting.css [hunk 3]
```diff
@@ -0,0 +1,18 @@
+.singleBookWrapper {
+    display: flex;
+    flex-direction: column;
+    align-content: center;
+    align-items: center;
+    /* border: solid 1px; */
```
Comment (Geoff10) [inline src/components/Booklisting/Booklisting.css:6]: Should: remove this commented out code if it is not being used.

File: src/components/Booklisting/Booklisting.jsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+// booklisting will take props to display JSON data from Home 
+function BookListing({title, author, image, genre}){
+
+    return(
+        <div className="singleBookWrapper">
+        <h1>{title}</h1>
+        <h2>{author}</h2>
+        <img src={image} alt={title}/>
+        <p>{genre}</p>
+        </div>
```
Comment (Geoff10) [inline src/components/Booklisting/Booklisting.jsx:11]: Should: fix the indentation here to imporove code readability.

File: src/components/Error/Error.jsx [hunk 5]
```diff
@@ -0,0 +1,8 @@
+function Error() {
+    return (
+        <>
```
Comment (Geoff10) [inline src/components/Error/Error.jsx:3]: Should: the fragment is not needed as there is only a single root element.

File: src/App.jsx [hunk 6]
```diff
@@ -1,34 +1,23 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
+import { BrowserRouter, Route, Routes } from 'react-router-dom'
 import './App.css'
-
+import Nav from './components/Nav/Nav'
+import Home from './components/home/home'
+import Error from './components/Error/Error'
 function App() {
```
Comment (Geoff10) [inline src/App.jsx:6]: Should: Put a newline after the imports.

File: src/components/Booklisting/Booklisting.jsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import "./Booklisting.css"
+// booklisting will take props to display JSON data from Home
```
Comment (Geoff10) [inline src/components/Booklisting/Booklisting.jsx:2]: Should: Put a newline after the imports.

File: src/components/Home/Home.css [hunk 8]
```diff
@@ -0,0 +1,8 @@
+.books-container {
+    display: flex;
+    flex-wrap: wrap;
+    gap: 10px;
+    justify-content: space-around;
+    margin-top: 300px;
+
```
Comment (Geoff10) [inline src/components/Home/Home.css:7]: Should: This line is not needed.

File: src/components/Home/Home.jsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+                setBooks(bookData.data)
+            })
+    }, [])
+   
+    return (
+        <>
+        
+        <div className='books-container'>
+            {books.map(book => 
+            <BookListing title={book.title} author={book.author} image={book.image} genre={book.genre.name} />
```
Comment (Geoff10) [inline src/components/Home/Home.jsx:24]: Must: The `key` attribute is missing here and causing a console error.

File: src/components/Home/Home.jsx [hunk 10]
```diff
@@ -0,0 +1,33 @@
+import "./Home.css"
+import { useEffect, useState } from "react"
+import BookListing from "../Booklisting/Booklisting"
+
+function Home() {
+    const [books, setBooks] = useState ([])
+
```
Comment (Geoff10) [inline src/components/Home/Home.jsx:7]: Should: remove the extra newline here.

File: src/components/Home/Home.jsx [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+            .then(function (res) {
+                return res.json()
+            })
+            .then(function (bookData) {
+                setBooks(bookData.data)
+            })
+    }, [])
+   
+    return (
+        <>
```
Comment (Geoff10) [inline src/components/Home/Home.jsx:20]: Should: The fragment is not needed here as the `div` is the only root element.

File: src/components/Home/Home.jsx (hunk 9 above)
Comment (Geoff10) [inline src/components/Home/Home.jsx:24]: Should: indentation for readability.

File: src/components/Nav/Nav.jsx [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import { NavLink } from "react-router-dom"
+import './Nav.css'
+
+function Nav() {
+    return (
+        <>
```
Comment (Geoff10) [inline src/components/Nav/Nav.jsx:6]: Should: fragment is not needed here as there is only one root element.

File: src/components/Nav/Nav.css [hunk 13]
```diff
@@ -0,0 +1,23 @@
+nav {
```
Comment (Geoff10) [inline src/components/Nav/Nav.css:1]: The content is visible behing the navbar making the items in the navbar difficult to read.
image

Comment (Geoff10) [inline src/components/Nav/Nav.css:1]: On smaller screens, the navbar is quite large and the "Book swap" text is cut off.
image

**Review by Geoff10 — CHANGES_REQUESTED** [review]
Good work, I left a few small comments.

**Comment by shellycs50** [comment]
Completed all requested changes in the Story 2 branch.

**Pull request merged**